### PR TITLE
Add drm_go2_ctx for odroidgo2

### PR DESCRIPTION
**Pull request by valadaa48**
- Revert odroidgo changes to drm_ctx
- Introduce new drm_go2_ctx specifically for odroidgo2 (and libgo2)
- Add hw_scaling option to mitigate builtin gl scaling/filtering and to
use libgo2's hw scaling and bicubic filtering via librga. The quality of
the image is improved by this and should be a bit faster.

Credit to CrashOverride for the initial libgo2 port and commit to RA.
Credit to natinusala for the initial cleanup/commit in the original drm_ctx driver.

@natinusala

### CODE REVIEW

File: menu/menu_setting.c [hunk 1]
```diff
@@ -6640,7 +6640,8 @@ void general_write_handler(rarch_setting_t *setting)
          }
          break;
       case MENU_ENUM_LABEL_VIDEO_SMOOTH:
-         video_driver_set_filtering(1, settings->bools.video_smooth);
+      case MENU_ENUM_LABEL_VIDEO_HW_SCALING:
+         video_driver_set_filtering(1, settings->bools.video_hw_scaling, settings->bools.video_hw_scaling);
```
Comment (RobLoach) [inline menu/menu_setting.c]: The `settings->bools.video_smooth` setting is now completely ignored. Would it be better to just re-purpose that setting?

Comment (valadaa48) [reply menu/menu_setting.c]: My first proof of concept indeed used used hw scaling when video_smooth was false but there are some cases of my testers wanting non-hw scaling (which implies bicubic).

**Comment by natinusala** [comment]
Thanks for the PR!

There is a reason why I merged the go2 port into the existing DRM driver, it's because I hate duplicating code. Now if there's a change to make in DRM we need to do it in both files... and I _know_ people won't do it and the go2 driver will be left behind.

Is there any reason to move the driver back in its own file?

Also, Rob is right here, I don't want any loss of features, especially about scaling since people are always so inquisitive about it.

**Comment by valadaa48** [comment]
I felt the merged ctx was difficult to navigate and even rebase with my branches given the sheer density of ifdef's. I can only imagine this alienating the other drm_ctx authors as well. I'm against code duplication like you but I think the drm_ctx code is simple and small enough to warrant it's own file. 

There is no loss of features here. There are 4 possibilities:

1. no bilinear + no hw scaling -> same as previous no bilinear
2. bilinear + no hw scaling -> same as previous bilinear
3. no bilinear + hw scaling -> hw scaling + bicubic
4. bilinear + hw scaling -> hw scaling + bicubic (bilinear is ignored)

I agree having a separate toggle may be a bit confusing which is why I mentioned having a 3 state toggle (for 1-3 above) might make more sense but the impact would be quite high.

**Comment by natinusala** [comment]
After Discord tests and discussion:

1. Add integer scaling support
2. Rename the setting `ctx_scaling` to make it more universal but keep it off by default
3. Fix the core override
4. Maybe use an ifdef to have the menu option shown as "RGA scaling" on odroidgo2? Also add "may break widgets" in the sublabel?
5. Maybe use the video smooth setting to have either bicubic or linear scaling in the go2 ctx driver?

**Comment by valadaa48** [comment]
> Add integer scaling support
This would require a lot of work for little benefit to only a few cores since the native res is only 480x320. I say we leave this as a fun exercise for later.

> Rename the setting ctx_scaling to make it more universal but keep it off by default
Done

> Fix the core override
Could possibly use some help here. I don't know why it's not working.

> Maybe use an ifdef to have the menu option shown as "RGA scaling" on odroidgo2? Also add "may break widgets" in the sublabel?
Done

> Maybe use the video smooth setting to have either bicubic or linear scaling in the go2 ctx driver?
Some more testing should be done before embarking on converting this bool to an enum (not to mention all the menu labels and sublabels.) Bicubic we know is superior in many cases but the other settings may be marginal.

**Comment by twinaphex** [comment]
OK, so two things -

there is a merge conflict, that has to be resolved first.

Second, @natinusala and others - is there a consensus as to how we are going to move this PR forward?

**Comment by natinusala** [comment]
We discussed it on Discord I think, it just needs to be updated

**Comment by valadaa48** [comment]
Rebased with master, @NOirBRight confirmed it saves by core now (:woman_shrugging: )

**Comment by twinaphex** [comment]
OK @natinusala , you feel this is fit to merge?

**Comment by natinusala** [comment]
If core overrides work now then sure

**Comment by twinaphex** [comment]
Does it work? Who can confirm that?

**Comment by valadaa48** [comment]
NOirBRight confirmed it works

**Comment by twinaphex** [comment]
OK, merging then.

**Pull request merged**